### DISCUSSION TITLE
Many to Many Relationship Entries, Add Relationship from Form Button

### DISCUSSION THREAD

**Discussion by Zero2XIII** [Budibase/budibase#7888] [Help] [answered]
Hi Budibase peoples,

I am having a hard time creating a many to many relationship entry from a form button.
Currently I have the following relationship set (Budibase Tables):
**Table_Vehicle**
- Job Card Link

**Table_Job_Card**
- Vehicle Link

I have a table with a button for each line of the table, the button function writes the id of the job card into the foreign key field of the Vehicle and it works to establish the relationship, however what I need to be able to do, it adding multiple relationships.

![1](https://user-images.githubusercontent.com/114136925/191682533-7208a94c-7004-46d6-88d3-abaaad9c25cb.PNG)

How can this be done? I am not finding this specific case scenario anywhere.

The two following cases must work:
1. From the job card, multiple vehicles must be linked.
2. Each vehicle must be linkable to multiple job cards.

The workflow will always be from the job card adding vehicles to the job card.
Previous relationships should never be broken, new relationships should always be added.

What I have tried on the save row function of the button, the following:
`{{ URL.jobcard_id }}` This over writes the relationship, but it does correctly link
`{{ URL.jobcard_id }} {{ Vehicle Tracking Table.Vehicle Tracking.Job Card Link }}` This over writes the relationship, but does correctly link
`{{ Vehicle Tracking Table.Vehicle Tracking.Job Card Link }} {{ URL.jobcard_id }}` This over writes the relationship, but clears it completely (Losing previous link)

![2](https://user-images.githubusercontent.com/114136925/191682538-97dfd52e-eb91-4e20-b9ca-a869a70954bc.PNG)

I have also tried the "JOIN" helper but to no avail. I have also tried adding commas between the two parts above and still nothing.

I'm hoping I am just being silly and am missing the simplest of things.
Any help will be appreciated.
Philip

EDIT:
Important note, the built in relationship pickers work flawlessly. I want a better user experience than what they can provide.

**Comment by melohagan** [comment]
Hey @Zero2XIII 

Would you be able to share your app export to support@budibase.com please?

**Reply by Zero2XIII** [reply]
HI Melohagan,

I can, but I had to remove the functionality and use a somewhat workaround so I have the functionality.
Can I do a mock-up and send that through?

Cheers,
Philip

**Reply by melohagan** [reply]
Certainly, that'd be helpful!

**Reply by Zero2XIII** [reply]
Hi,

It was a very quick and dirty mock-up but I think the "issue" shows.
I am uploading it here just for references as well.

Best regards,
Philip
[Sample Many to Many-export-1664364726723.txt](https://github.com/Budibase/budibase/files/9664453/Sample.Many.to.Many-export-1664364726723.txt)

**Comment by melohagan** [comment, marked as answer]
Hey @Zero2XIII 

This isn't too difficult to do. The relationships are stored as an array, so you just need to push the new relationships.

Link Job Card actions:
![Screenshot 2022-09-29 at 10 32 56](https://user-images.githubusercontent.com/101575380/192996354-96dbb1f3-812e-4d41-a4e1-470f1bd21078.png)

Binding:
```js
let links = $("Vehicle Repeater.Vehicle.Job Card Link");
if (links?.length > 0) {
	links.push($("Job Card Table.Job Card._id"))
}
else {
	links = [ $("Job Card Table.Job Card._id") ]
}
return links;
```

Here's the app export: 
[Many to Many Zero 2-export-1664444119430.txt](https://github.com/Budibase/budibase/files/9673299/Many.to.Many.Zero.2-export-1664444119430.txt)

**Result**
![many-links](https://user-images.githubusercontent.com/101575380/192997168-bc1f3d1f-99be-4a0e-99de-b92326687731.gif)

**Reply by Zero2XIII** [reply]
Fantastic! Thank you so much. This solved the issue.

**Reply by melohagan** [reply]
You're welcome! 😄
